### ISSUE TITLE
Generator produces an invalid enum with an underscore "_" as the name

### ISSUE THREAD

**Issue by nyoungac** [jOOQ/jOOQ#10866]
Using the 3.14.1 code generator with Java on a MySQL database with an enum field defined as:

```sql
  `t_offset_o` enum('-','+') NOT NULL DEFAULT '+',
```

OR

```sql
  `ref` enum('', <...>) NOT NULL DEFAULT '',
```

produce an invalid underscore "_" enum:

    _(""),

expected: Have some placeholder or random name.

**Comment by lukaseder** [comment]
Thanks for your report. As a workaround, you can always implement a generator strategy to assign different identifiers to your enums: https://www.jooq.org/doc/latest/manual/code-generation/codegen-generatorstrategy/

**Comment by lukaseder** [comment]
This seems to be an oversight in the fix for #4703.

**Comment by lukaseder** [comment]
Hmm, no. Unfortunately, generator strategies aren't involved in generating the identifiers for enum literals. That might be another issue: https://github.com/jOOQ/jOOQ/issues/10881

Your examples show an interesting case. While it would be possible to disambiguate the `_` characters individually, if you had created an enum like this, we'd get yet another issue where disambiguated identifiers create a conflict, because they're both the same:

```sql
enum ('', '-')
```

I don't want to rush fixing this. It seems that name disambiguation could use some more thorough thought...

**Comment by lukaseder** [comment]
Another workaround is to use a `CHECK (t_offset_o IN ('-', '+'))` constraint, and make it `CHAR(1)` instead of `ENUM`

**Comment by berndwaibel** [comment]
Same Problem here: MySQL, Enum Type:
enum('store','single','multiple','sold','')
The "" string does produce an underscore.

This is a very bad problem, as this is no more supported in JDK > 9.
So the created code (created by Jooq 3.14.4) will not compile if the JDK Level is 9 or above

**Comment by lukaseder** [comment]
Starting from jOOQ 3.16, such problems can be worked around using a `GeneratorStrategy` configuration: https://github.com/jOOQ/jOOQ/issues/10881

We should obviously never generate `_` enum literals, but at least there will be a workaround until this is fixed.

**Comment by lukaseder** [comment]
I'm fixing this particular issue here, where `enum` literals can be produced by the name `_`. For now, they just generate `__`. It seems that the fix from #4703 wasn't applied to enum literals.

There's still more name mangling to be done, but it might be more edge-casey than this one (?): https://github.com/jOOQ/jOOQ/issues/12732

Again, at least with 3.16, the `GeneratorStrategy` now supports overriding the default behaviour for enum literals as well via #10881.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16. I won't backport this fix, because it could break existing generated code elsewhere.
